packages/scripts/src/materialize/materializedViewUtils.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  Db,
+  IndexSpecification,
+  Collection,
+  CreateIndexesOptions,
+} from "mongodb-rag-core";
+
+export type IndexDefinition = {
+  spec: IndexSpecification;
+  options?: CreateIndexesOptions;
+  name?: string;
+};
+
+export type EnsureCollectionWithIndexParams = {
+  db: Db;
+  collectionName: string;
+} & ({ index: IndexDefinition } | { indexes: IndexDefinition[] });
+
+export async function ensureCollectionWithIndex({
+  db,
+  collectionName,
+  ...indexProps
+}: EnsureCollectionWithIndexParams) {
+  console.log("Ensuring collection with index:", {
```

Comment:
should be `logger.info()` i think to be consistent w our other stuff

Reply:
ditto below `console.log`s

Reply:
We don't have a logger library for the `scripts` directory, lots of `console.log()` in this package